### ISSUE TITLE
Azure Container Instance Operator uses deprecated "network profile" arg

### ISSUE THREAD

**Issue by jackglick** [apache/airflow#34820]
### Apache Airflow version

2.7.1

### What happened

The Azure Container Instance Operator uses the "network profile" argument which is since deprecated by Microsft: https://learn.microsoft.com/en-us/azure/container-instances/container-instances-vnet

Azure Container Instances deployed by the Operator are not deployed into a virtual network or subnet.

### What you think should happen instead

Get rid of the network profile argument and replace with "vnet" and "subnet" arguments as detailed in this documentation: https://learn.microsoft.com/en-us/azure/container-instances/container-instances-vnet

### How to reproduce

Deploy azure container instance - there are no longer network profiles supported by Azure. Will not let you deploy into your virtual network or subnet.

### Operating System

linux

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Lee-W** [comment]
https://github.com/apache/airflow/pull/34738 is now working on it. I think it will be included in the next azure provider release

**Comment by eladkal** [comment]
@pankajastro I assume this issue is fixed in https://github.com/apache/airflow/pull/34738 ?

**Comment by pankajastro** [comment]
> @pankajastro I assume this issue is fixed in #34738 ?

I removed network_profile in https://github.com/apache/airflow/pull/34738 but just created a separate PR https://github.com/apache/airflow/pull/34850 to add subnet_ids

**Comment by pankajastro** [comment]
> > > @pankajastro I assume this issue is fixed in #34738 ?
> > 
> > 
> > I removed network_profile in #34738 but just created a separate PR #34850 to add subnet_ids
> 
> Do we also need to add in a virtual network param?

the subnet is something like and can't see specific param for vnet in SDK

```
"subnetIds": [{
    "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('subnetName'))]"
}],
```

**Comment by eladkal** [comment]
closing as task is completed and it seems more like a support question.
If further support is needed please use Github Discussion / Stackoverflow
